src/network/stream_reader.rs

```diff
@@ -47,34 +53,59 @@ impl<R: Read> fmt::Debug for StreamReader<R> {
 impl<R: Read> StreamReader<R> {
     /// Constructs new stream reader for a given input stream `stream` with
     /// optional parameter `buffer_size` determining reading buffer size
-    pub fn new(stream: R, buffer_size: Option<usize>) -> StreamReader<R> {
+    /// Specify `read_until` bytes when expecting big objects coming from this reader. If not
+    /// specified it's 0, trying to parse `Decodable` every received bytes. Using the default (0)
+    /// when parsing a big object like a full block result in poor performance due to the repeated
+    /// parsing attempts on incomplete data.
```

Comment:
In case you want to use it, I massaged the docs a bit while understanding them.
```
    /// Construct a stream reader for a given input `stream`. Use optional parameter `buffer_size`
    /// to control the read buffer size (default: 64K). Specify `read_until` bytes when expecting
    /// big objects coming from this reader. If not specified it's 0, trying to parse `Decodable`
    /// every received bytes. Using the default (0) when parsing a big object like a full block can
    /// result in poor performance due to the repeated parsing attempts on incomplete data.
```